### PR TITLE
Add a local typedef IndexSet::value_type.

### PR DESCRIPTION
**Pull request by bangerth**
We'll need this for making FEValues::get_function_values work for arbitrary scalar types.

### CODE REVIEW

File: doc/news/changes.h [hunk 1]
```diff
@@ -350,6 +350,11 @@ inconvenience this causes.
 <h3>Specific improvements</h3>
 
 <ol>
+  <li> New: IndexSet now has a local typedef IndexSet::value_type/
```
Comment (tjhei) [inline doc/news/changes.h]: what is the deal with the slash at the end? Typo?

**Comment by tjhei** [comment]
Not sure where this is going, but it looks reasonable.

File: doc/news/changes.h (hunk 1 above)
Comment (bangerth) [inline doc/news/changes.h]: Yes, typo. Fixed.

**Comment by bangerth** [comment]
The destination is that we currently support
```
  IndexSet active_set;
  data_out.add_vector (active_set);
```
and similar things where one considers an IndexSet as a vector of values that are either true or false. To make this work better in a generic way, I just needed the same typedef that we have in all other vectors as well.

**Comment by tjhei** [comment]
Okay, got it.

**Pull request merged**